File: DuplicatedFragmentTest/app/src/main/java/charlezz/com/duplicatedfragmenttest/MainActivity.java
package charlezz.com.duplicatedfragmenttest;

import android.Manifest;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.gun0912.tedpermission.PermissionListener;
import com.gun0912.tedpermission.TedPermission;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        new TedPermission(this)
                .setPermissions(Manifest.permission.WRITE_EXTERNAL_STORAGE)
                .setPermissionListener(new PermissionListener() {
                    @Override
                    public void onPermissionGranted() {
                        getSupportFragmentManager().beginTransaction().add(R.id.container, new VideoFragment()).commit();
                        UIFragment f = new UIFragment();
                        getSupportFragmentManager().beginTransaction().add(R.id.container,f ).commit();
                    }

                    @Override
                    public void onPermissionDenied(ArrayList<String> deniedPermissions) {
                        finish();
                    }
                })
                .check();


    }
}

File: IPCExample/App1/src/main/java/net/ddns/oksisi2/app1/MainActivity.kt
package net.ddns.oksisi2.app1

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log

class MainActivity : AppCompatActivity() {

    val TAG = MainActivity::class.java.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.e(TAG,"onCreate")
    }
}
